src/Field/Definition/ChoiceField.php

```diff
@@ -110,7 +110,7 @@ private function validateSingleSelect (ComponentContext $context, array $content
 			],
 		);
 
-		\assert(null === $data || \is_int($data) || \is_string($data));
+		\assert(\is_int($data) || \is_string($data));
```

Comment:
This assertion is incorrect for optional fields

Reply:
And you could do an early exit here if it is null.

So maybe leave the assertion as is and add the early exit beforehand.

Reply:
The assertion for a single-option field is correct. An optional, non-filled field will result in an empty string.